Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/nbclient/client.py", line 1117, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/nbclient/util.py", line 78, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/nbclient/util.py", line 57, in just_run
    return loop.run_until_complete(coro)
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/nbclient/client.py", line 558, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/nbclient/client.py", line 862, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/nbclient/client.py", line 765, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
data.groupby('state')['unemployment_rate'] \
        .aggregate({'Min. Unemploy.' : np.min,
                    'Mean Unemploy.' : np.mean,
                    'Max. Unemploy.' : np.max})
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mSpecificationError[0m                        Traceback (most recent call last)
[0;32m/tmp/ipykernel_2921/3467181496.py[0m in [0;36m<module>[0;34m[0m
[0;32m----> 1[0;31m [0mdata[0m[0;34m.[0m[0mgroupby[0m[0;34m([0m[0;34m'state'[0m[0;34m)[0m[0;34m[[0m[0;34m'unemployment_rate'[0m[0;34m][0m[0;31m [0m[0;31m\[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      2[0m         .aggregate({'Min. Unemploy.' : np.min,
[1;32m      3[0m                     [0;34m'Mean Unemploy.'[0m [0;34m:[0m [0mnp[0m[0;34m.[0m[0mmean[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m      4[0m                     'Max. Unemploy.' : np.max})

[0;32m/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/pandas/core/groupby/generic.py[0m in [0;36maggregate[0;34m(self, func, engine, engine_kwargs, *args, **kwargs)[0m
[1;32m    247[0m             [0;31m# but not the class list / tuple itself.[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[1;32m    248[0m             [0mfunc[0m [0;34m=[0m [0mmaybe_mangle_lambdas[0m[0;34m([0m[0mfunc[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 249[0;31m             [0mret[0m [0;34m=[0m [0mself[0m[0;34m.[0m[0m_aggregate_multiple_funcs[0m[0;34m([0m[0mfunc[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    250[0m             [0;32mif[0m [0mrelabeling[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m    251[0m                 [0;31m# error: Incompatible types in assignment (expression has type[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m

[0;32m/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/pandas/core/groupby/generic.py[0m in [0;36m_aggregate_multiple_funcs[0;34m(self, arg)[0m
[1;32m    282[0m             [0;31m# have not shown a higher level one[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[1;32m    283[0m             [0;31m# GH 15931[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 284[0;31m             [0;32mraise[0m [0mSpecificationError[0m[0;34m([0m[0;34m"nested renamer is not supported"[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    285[0m [0;34m[0m[0m
[1;32m    286[0m         [0;32melif[0m [0many[0m[0;34m([0m[0misinstance[0m[0;34m([0m[0mx[0m[0;34m,[0m [0;34m([0m[0mtuple[0m[0;34m,[0m [0mlist[0m[0;34m)[0m[0;34m)[0m [0;32mfor[0m [0mx[0m [0;32min[0m [0marg[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;31mSpecificationError[0m: nested renamer is not supported
SpecificationError: nested renamer is not supported

